### ISSUE TITLE
Client state doesn't change after disconnection

### ISSUE THREAD

**Issue by YohskDista** [grpc/grpc-dotnet#1885]
### What version of gRPC and what language are you using?

Grpc.Net.Client 2.48.0 in C#

### What operating system (Linux, Windows,...) and version?

Windows 10 build 19044

### What runtime / compiler are you using (e.g. .NET Core SDK version `dotnet --info`)

.NET 6.0

### What did you do?

I created a client with gRPC .NET that connects to a gRPC Core server. Here is the step that I've made during my tests:

* The gRPC .NET client connects to the gRPC Core server
* When the server is closed the client is not notified that the connection is interrupted. 
* The `ConnectivityState` is always in `Ready` state.

I tried to create a watchdog on a separate thread that will use the `WaitForStateChangedAsync` method but it waits undefinetly because of the not changing state. I call this method like this:

`await _channel.WaitForStateChangedAsync(_channel.State).ConfigureAwait(false);`

I also tried to create a loop like this:

```
while (_channel.State == ConnectivityState.Ready)
{
     _logger.LogDebug("The channel state is still connected");
     Task.Delay(500);
}

_logger.LogDebug("------- Oups, the connection is closed!!!!");
```

But I never get out of the loop.

I noticed that the reconnection is correctly made, there is no problem on that. It's just that we are not aware of the client disconnection. Maybe I am missing some configuration but I didn't find any explanations about this on StackOverflow or in the previous issues.

### What did you expect to see?

I expect that the previous code will continue when a client is disconnected from the server.

Thanks in advance!

**Comment by JamesNK** [comment]
Are you customizing the HttpHandler in the gRPC options? If you do then for the state updates to work, it must be a SocketsHttpHandler with no ConnectCallback customized.

**Comment by YohskDista** [comment]
No I am not because of the `WaitForStateChangedAsync` XML documentation explaining that it works only with `SocketsHttpHandler` (default value). Here is the code that I use to create the `GrpcChannel`:

```csharp
_channel = GrpcChannel.ForAddress(
            $"http://{serverConfiguration.ServerAddress}:{serverConfiguration.ServerPort}",
            new GrpcChannelOptions
            {
                Credentials = ChannelCredentials.Insecure,
            });
```

**Comment by YohskDista** [comment]
They are on the same machine. Do you suggest to use the inter-process instead of the HTTP connection?

**Comment by YohskDista** [comment]
Maybe a few more information about what I tried last day:

The gRPC client subscribes to gRPC stream to be notified when a value as changed on server side. When the server software is closed normaly we are not notified that the channel state changed.
I tried to create an observable like:

```csharp
Observable.Interval(TimeSpan.FromSeconds(5))
          .Subscribe(_ => _logger.LogDebug("The state of the channel connection is {State}", _channel.State));
```

But the state displays always `Ready`, even if the server is shutdown.

I've also tried to configure an HttpHandler in the `GrpcChannelOptions` like this:

```csharp
HttpHandler = new SocketsHttpHandler
{
      KeepAlivePingDelay = TimeSpan.FromSeconds(5),
      KeepAlivePingPolicy = HttpKeepAlivePingPolicy.Always
}
```

But nothing changed, the channel state has always the `Ready` value
